test/disc_test.rb

```diff
@@ -85,68 +104,32 @@
   end
 
   test 'jobs are executed' do
-    begin
-      Echoer.enqueue(['one argument', { random: 'data' }, 3])
+    Echoer.enqueue(['one argument', { random: 'data' }, 3])
 
-      cout, _, pid = PTY.spawn(
-        'QUEUES=test,default ruby -Ilib bin/disc -r ./examples/echoer'
-      )
-      sleep 0.5
+    run('QUEUES=test ruby -Ilib bin/disc -r ./examples/echoer') do |cout, pid|
+      output = cout.take(3)
```

Comment:
Since `IO#gets` blocks, asking for the first 3 lines of output will effectively wait *exactly* as much as we want it to wait, and not a millisecond longer.